docs/06-tutorials/02-todolist.md

```diff
@@ -0,0 +1,477 @@
+    const { hash } = await todoListContract.toggleCompleted(taskId);
+    await provider.waitForTransaction(hash);
+    await getTasks();
+  };
+
+
+  return (
+    <div className="App">
+      <h1>Todo List</h1>
+
+
+      <form onSubmit={createTask}>
+        <input
+          type="text"
+          placeholder="Add a Task"
+          value={taskContent}
+          onChange={(e) => setTaskContent(e.target.value)}
+        />
+        <button type="submit">Add</button>
+      </form>
+
+
+      <div>
+        {tasks.map((task) => (
+          <div key={task.id}>
+            <input
+              type="checkbox"
+              checked={task.completed}
+              onChange={() => toggleCompleted(task.id)}
+            />
+            <span>{task.content}</span>
+          </div>))
+        }
+      </div>
+    </div>
+  );
+}
+
+
+export default App;
+```
+
+11. Test the dApp creation by using command:
+
+```
+npm start
+```
+
+   **Note:** Remember to manually connect the todolist dApp to your wallet. There is no verification that the Gobi network is using the correct dApp.
+
+## Playing with the dApp
+
+The todolist dApp is displayed on your web browser (such as Chrome) using a localhost port. 
+It is now functional. Play with the dApp by adding tasks and checking them off.
+
+   **Note:** If the dApp does not appear in your browser, then check the setting in your browser to allow **JavaScript** to be enabled. Also, make sure that your MetaMask wallet with the Gobi Testnet network is connected to the web browser.
+
+<p>
+<img src={require("/img/docs/tutorial/todolist-dApp.png").default} alt="remix-compiler" width="300" height="200" />
+</p>
+
+### Optional: Connection Fix
+
+Use the following steps only if you are not able to add tasks and connect MetaMask to the Todo List dApp.
+
+In the App.js file, use vi editor to perform the following:
+
+1. After the line:
+
+```
+ function App(){
+```
+   Add:
+
+```
+const [isConnected, setConnected] = useState(false);
+```
+
+   **Note:** This line defines a state **isConnected** that updates with the **setConnected** modifier and sets to **false**.
+
+2. Before the line:
+
+```
+const getTasks = async () => {
+```
+   Add:
+
+```
+const tryConnect = async () => {
+	const ret = await provider.send('eth_requestAccounts', []);
+	await getTasks();
+	setConnected(ret !== false);
+  }
+```
+
+   **Note:** When **tryConnect** is called to connect to the local MetaMask accounts, it loads the **todo tasks** and updates the **connection status**.
+
+3. Before the line:
+
+```
+ return (
+	<div className="App">
+```
+   Add:
+
+```
+if (!isConnected) return <div className="App"><button onClick={tryConnect}>Connect</button></div>
+```
+
+   **Note:** If the local MetaMask account is not connected, then this code snippet adds a button. This button calls **tryConnect** when it is clicked. If connected, the following code executes itself.
+
+4. Save the updates.
+
+
+
+
+
+
+
+
+

```

Comment:
Can we clean this up?

```suggestion
```

Also, I think it might be helpful if we add the complete code as step 5 so the user knows exactly what the suggested changes are.